opts/options.go

```diff
@@ -38,8 +28,16 @@ func RoutingTableRefreshPeriod(period time.Duration) dht.Option {
 // Deprecated: use dht.Datastore
 func Datastore(ds ds.Batching) dht.Option { return dht.Datastore(ds) }
 
-// Deprecated: use dht.Client
-func Client(only bool) dht.Option { return dht.Client(only) }
+// Client configures whether or not the DHT operates in client-only mode.
+//
+// Defaults to false (which is ModeAuto).
+// Deprecated: use dht.Mode(ModeClient)
+func Client(only bool) dht.Option {
+	if only {
+		return dht.Mode(dht.ModeClient)
+	}
+	return dht.Mode(dht.ModeAuto)
```

Comment:
Is making the default here ModeAuto instead of doing nothing when `only ==false` ok?